tests/test_polling_manager.py

```diff
@@ -35,3 +38,37 @@ def test_polling_manager_calls_update_environment_on_each_refresh():
     # for each subsequent refresh
     assert flagsmith.update_environment.call_count == 3
     polling_manager.stop()
+
+
+def test_polling_manager_is_resilient_to_api_errors(server_api_key):
+    with mock.patch("requests.Session") as session_mock:
+        # Given
+        session_mock.return_value = mock.MagicMock(get=mock.MagicMock(status_code=500))
```

Comment:
Let's construct the mock in a more readable way:

```suggestion
        session_mock.get.return_value = mock.MagicMock(status_code=500)
```

Additionally, consider using `pytest-mock`'s `mocker` fixture (see [docs](https://pytest-mock.readthedocs.io/en/latest/usage.html#usage)).

Reply:
What a silly approach I had there. Easy to tell I went over a couple of iterations. Thanks! I'll move over to `mocker` for consistency. 👍